src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -294,10 +294,10 @@ internal IInternalActorRef ResolveActorRefWithLocalAddress(string path, Address
                     // HACK: needed to make ActorSelections work
                     if (actorPath.ToStringWithoutAddress().Equals("/"))
                         return RootGuardian;
-                    return _local.ResolveActorRef(RootGuardian, actorPath.Elements);
+                    return _local.ResolveActorRef(RootGuardian, actorPath.ElementsWithUid);
                 }
                     
-                return new RemoteActorRef(Transport, localAddress, new RootActorPath(actorPath.Address) / actorPath.Elements, ActorRefs.Nobody, Props.None, Deploy.None);
+                return new RemoteActorRef(Transport, localAddress, new RootActorPath(actorPath.Address) / actorPath.ElementsWithUid, ActorRefs.Nobody, Props.None, Deploy.None);
```

Comment:
All three of these `ActorPath.ElementsWithUid` calls preserve the UID of the actor whose reference we are trying to resolve on the very last element in the `IEnumerable<string>`. Needed to guarantee that UIDs are respected for actor references.